pkg/auxnetdevice/auxNetDeviceProvider.go

```diff
@@ -0,0 +1,181 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package auxnetdevice
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/jaypipes/ghw"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/types"
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/utils"
+)
+
+type auxNetDeviceProvider struct {
+	deviceList []*ghw.PCIDevice
+	rFactory   types.ResourceFactory
+}
+
+// NewAuxNetDeviceProvider DeviceProvider implementation from auxNetDeviceProvider instance
+func NewAuxNetDeviceProvider(rf types.ResourceFactory) types.DeviceProvider {
+	return &auxNetDeviceProvider{
+		rFactory:   rf,
+		deviceList: make([]*ghw.PCIDevice, 0),
+	}
+}
+
+func (ap *auxNetDeviceProvider) GetDiscoveredDevices() []*ghw.PCIDevice {
+	return ap.deviceList
+}
+
+func (ap *auxNetDeviceProvider) GetDevices(rc *types.ResourceConfig) []types.HostDevice {
+	newAuxDevices := make([]types.HostDevice, 0)
+	for _, device := range ap.deviceList {
+		// discover auxiliary device names
+		auxDevs, err := utils.GetAuxNetDevicesFromPci(device.Address)
+		if err == nil {
+			if len(auxDevs) == 0 {
+				glog.Warningf("auxnetdevice GetDevices(): no auxiliary devices for PCI %s", device.Address)
+				continue
+			}
+			for _, auxDev := range auxDevs {
+				if newDevice, err := NewAuxNetDevice(device, auxDev, ap.rFactory, rc); err == nil {
+					newAuxDevices = append(newAuxDevices, newDevice)
+				} else {
+					glog.Warningf("auxnetdevice GetDevices(): error creating new device %s PCI %s: %q",
+						auxDev, device.Address, err)
+				}
+			}
+		} else {
+			glog.Warningf("auxnetdevice GetDevices(): error getting auxnetdevices from device %s: %q",
```

Comment:
nit: auxNetDevice ? here and below at L77

Reply:
Ok